unwind/src/lib.rs

```diff
@@ -96,45 +135,65 @@ struct UnwindInfo<R: Reader> {
     personality: Option<Pointer>,
     lsda: Option<Pointer>,
     initial_address: u64,
-    ctx: UninitializedUnwindContext<EhFrame<StaticReader>, StaticReader>,
 }
 
 impl ObjectRecord {
-    fn unwind_info_for_address(&self,
-                               ctx: UninitializedUnwindContext<EhFrame<StaticReader>, StaticReader>,
-                               address: u64) -> gimli::Result<UnwindInfo<StaticReader>> {
+    fn unwind_info_for_address(
+        &self,
+        ctx: &mut UninitializedUnwindContext<StaticReader>,
+        address: u64,
+    ) -> gimli::Result<UnwindInfo<StaticReader>> {
         let &ObjectRecord {
             ref eh_frame_hdr,
-            eh_frame: ref sel,
+            ref eh_frame,
             ref bases,
             ..
         } = self;
 
-        let fde = eh_frame_hdr.table().unwrap()
-            .lookup_and_parse(address, bases, sel.clone(), |offset| sel.cie_from_offset(bases, offset))?;
-
+        let fde;
         let mut result_row = None;
-        let mut ctx = ctx.initialize(fde.cie()).unwrap();
-
-        {
-            let mut table = UnwindTable::new(&mut ctx, &fde);
-            while let Some(row) = table.next_row()? {
-                if row.contains(address) {
-                    result_row = Some(row.clone());
-                    break;
+        use gimli::UnwindSection;
+
+        let mut entries = eh_frame.entries(bases);
+        backtrace::resolve(address as *mut _, |s| {
+            println!("address {:016x}: {:?}", address, s.name());
+            let (text, eh_frame) = match self.er { EhRef::WithoutHeader { text, eh_frame, .. } => (text, eh_frame), _ => panic!() };
+            println!("bases {:?}", bases);
+            println!("text {:016x} .. {:016x}", text.start, text.end);
+            println!("eh_frame {:016x} .. {:016x}", eh_frame.start, eh_frame.end);
+        });
+        let mut i = 0;
+        while let Some(entry) = entries.next()? {
+            match entry {
+                gimli::CieOrFde::Cie(_) => {}
+                gimli::CieOrFde::Fde(partial) => {
+                    let fde = partial.parse(EhFrame::cie_from_offset).unwrap();
+                    println!("fde {:016x} .. {:016x}", fde.initial_address(), fde.initial_address() + fde.len());
+                    //println!("{:?}", fde);
                 }
             }
+            i += 1;
+            if i > 10 {
+                break;
+            }
         }
 
+        fde = eh_frame.fde_for_address(bases, address, EhFrame::cie_from_offset).unwrap();
```

Comment:
It fails here

Reply:
This works on linux (after I hack it to use EhRef::WithoutHeader for linux too).